frame/dex-router/src/tests.rs

```diff
@@ -148,6 +161,40 @@ fn get_route_tests() {
 	});
 }
 
+#[test]
+fn halborn_hal11_route_with_cycle() {
+	new_test_ext().execute_with(|| {
+		let currency_pair = CurrencyPair { base: USDT, quote: USDC };
+		assert_eq!(DexRouter::get_route(currency_pair), None);
+
+		let dex_route =
+			vec![create_usdt_usdc_pool(), create_usdc_usdt_pool(), create_usdt_usdc_pool()];
```

Comment:
So this problem only happens if someone updates a cyclic route? Shouldn't we validate at the time of update then?

Reply:
so if want validation for this case then we can improve, or we just trust that users won't do such routes...

Reply:
I see, Is it hard to validate?

Reply:
no, I think we can use https://docs.substrate.io/rustdocs/latest/sp_std/collections/btree_set/struct.BTreeSet.html to detect duplicates. 
By tracking pool_id in set we can eliminate second case in the test. And by tracking (base,quote) pair in set we can eliminate both cases.

Reply:
I see, seems the second case would be eliminated with tracking base/quote in a set anyway.

Reply:
Yes that seem good way.